### PR TITLE
✨ Project statistic 

### PR DESCRIPTION
**Pull request by yipkhenlai**
# Description

New API was added to retrieve information on the number of labeled images, number of unlabeled images, and total number of labels per class in the project

API : http://localhost:{port}/v2/:annotation_type/projects/:project_name/statistic

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (Non-breaking change which fixes an issue)
- [x] New feature (Non-breaking change which adds functionality)
- [ ] Breaking change (Fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update (Example: REST API changes)

# Tested on?

- [x] Windows  
- [ ] Linux Ubuntu 
- [ ] Centos 
- [ ] Mac  
- [ ] Others  (State here -> xxx )  

# Checklist:

- [x] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged
- [x] CHANGELOG.md has been updated
- [ ] Update [Gitbook REST API Page](https://docs.classifai.ai/v/dev/development/rest-api)

### CODE REVIEW

File: classifai-core/src/main/java/ai/classifai/database/annotation/bndbox/BoundingBoxVerticle.java [hunk 1]
```diff
@@ -80,6 +82,10 @@ else if(action.equals(AnnotationQuery.getRenameProjectData()))
         {
             this.renameProjectData(message);
         }
+        else if(action.equals(AnnotationQuery.getRetrieveAllProjects()))
+        {
+            this.renameProjectData(message);
```
Comment (devenyantis) [inline classifai-core/src/main/java/ai/classifai/database/annotation/bndbox/BoundingBoxVerticle.java]: What is this for? retreive project data and then rename project data

File: classifai-core/src/main/java/ai/classifai/database/portfolio/PortfolioDbQuery.java [hunk 2]
```diff
@@ -58,4 +58,7 @@
     @Getter private static final String reloadProject = "SELECT project_path FROM Portfolio WHERE project_id = ?";
 
     @Getter private static final String updateLastModifiedDate = "UPDATE Portfolio SET current_version = ? WHERE project_id = ?";
+
+    @Getter private static final String RETRIEVEPROJECTSTATISTIC = "retrieveProjectStatistic";
```
Comment (devenyantis) [inline classifai-core/src/main/java/ai/classifai/database/portfolio/PortfolioDbQuery.java]: Although sonar cloud will complain, let's be consistent with our codebase. Camel case as before.

File: classifai-core/src/main/java/ai/classifai/util/ParamConfig.java [hunk 3]
```diff
@@ -95,6 +95,12 @@
     @Getter private static final String createdDateParam = "created_date";
     @Getter private static final String lastModifiedDate = "last_modified_date";
     @Getter private static final String isRootPathValidParam = "root_path_valid";
+    @Getter private static final String LABELEDIMAGEPARAM = "labeled_image";
```
Comment (devenyantis) [inline classifai-core/src/main/java/ai/classifai/util/ParamConfig.java]: This var as well. Consistent with the codebase

File: classifai-core/src/main/java/ai/classifai/util/ParamConfig.java [hunk 4]
```diff
@@ -95,6 +95,12 @@
...
+    @Getter private static final String UNLABELEDIMAGEPARAM = "unlabeled_image";
+    @Getter private static final String LABELPERCLASSINPROJECTPARAM = "label_Per_Class_In_Project";
```
Comment (devenyantis) [inline classifai-core/src/main/java/ai/classifai/util/ParamConfig.java]: all small letter after underscore

File: classifai-core/src/main/java/ai/classifai/util/data/LabelListHandler.java [hunk 5]
```diff
@@ -0,0 +1,200 @@
...
+        ArrayList<String> labels = new ArrayList<>();
+
+        for (int i = 0; i < labelPointData.size(); i++)
+        {
+            JsonObject jsonArray = labelPointData.getJsonObject(i);
+            String label = jsonArray.getString("label");
+            labels.add(label);
+        }
+
+        for (String label : labels)
```
Comment (devenyantis) [inline classifai-core/src/main/java/ai/classifai/util/data/LabelListHandler.java]: Can this process be inside the loop before. So we don't have to run loop twice.

File: classifai-core/src/main/java/ai/classifai/database/annotation/bndbox/BoundingBoxVerticle.java (hunk 1 above)
Comment (ken479) [reply classifai-core/src/main/java/ai/classifai/database/annotation/bndbox/BoundingBoxVerticle.java:87]: this is for retrieving the  project data for the database migration

File: classifai-core/src/main/java/ai/classifai/database/portfolio/PortfolioDbQuery.java (hunk 2 above)
Comment (yipkhenlai) [reply classifai-core/src/main/java/ai/classifai/database/portfolio/PortfolioDbQuery.java]: I see. Ok, i will follow back

File: classifai-core/src/main/java/ai/classifai/util/data/LabelListHandler.java (hunk 5 above)
Comment (yipkhenlai) [reply classifai-core/src/main/java/ai/classifai/util/data/LabelListHandler.java]: I will try

File: classifai-core/src/main/java/ai/classifai/database/annotation/bndbox/BoundingBoxVerticle.java (hunk 1 above)
Comment (yipkhenlai) [reply classifai-core/src/main/java/ai/classifai/database/annotation/bndbox/BoundingBoxVerticle.java]: @devenyantis wrongly inserted action has been removed. sorry for causing confusion.

**Review by devenyantis — APPROVED** [review]

**Pull request merged**